### ISSUE TITLE
Difference between NumericFormat component and numericFormatter function

### ISSUE THREAD

**Issue by thiengo105** [s-yadav/react-number-format#806]
#### Describe the issue and the actual behavior
`numericFormatter` function should return the same value as `NumberFormat` component with the same props.
![image](https://github.com/s-yadav/react-number-format/assets/39598876/ac7a4724-6882-407d-9b6b-dbb2ceb47244)

#### Describe the expected behavior
`numericFormatter` should return 3.91 instead of 3.90

#### Provide a CodeSandbox link illustrating the issue
https://gnsqpq-5173.csb.app/

#### Provide steps to reproduce this issue
use the following props to pass to `numericFormatter` function and `NumerFormat` component:
```
value = 3.9082;
decimalScale = 2;
```

#### Please check the browsers where the issue is seen
- [x] Chrome
- [ ] Chrome (Android)
- [ ] Safari (OSX)
- [ ] Safari (iOS)
- [ ] Firefox
- [ ] Firefox (Android)

**Comment by portare** [comment]
Also `numericFormatter ` doesnt apply `mapToRadix` property

**Comment by RifatF** [comment]
Same problem with version 5.4.0. The parameter `valueIsNumericString`  doesn't change anything.

**Comment by AI111** [comment]
Here is the reason https://github.com/s-yadav/react-number-format/blob/master/src/numeric_format.tsx#L478
NumericFormat  Component round value before formatting it with numericFormatter function

**Comment by YannickHeeftEenGeleFiets** [comment]
I agree with the expectation that this utility should behave the same as the component. Could the maintainers explain this difference in behaviour? Is this intended?

I'm considering fixing this behaviour on the consumption side, but I'd rather not since I'm using this very package to delegate this work to.

I'm also happy to contribute, but I would like to hear the stance of the maintainers before a potential pull request gets rejected. 🤷 

In the meantime, thanks in advance for your time :)
